### PR TITLE
homework express wk3

### PR DESCRIPTION
**Pull request by Nadoshka**

### CODE REVIEW

File: week3/homework/src/actions/clearTodos.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const saveTodos = './index';
```
Comment (gajduk) [inline week3/homework/src/actions/clearTodos.js]: I think you might be missing something :)

File: week3/homework/src/actions/clearTodos.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+function clearTodos(todos) {
+  return new Promise((resolve, reject) => {
```
Comment (gajduk) [inline week3/homework/src/actions/clearTodos.js]: Can this be simpler?

File: week3/homework/src/actions/createTodo.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    if (todo === null) {
+      reject(new Error('No todo set!'));
+    }
+    else {
+      if (todo.description !== null) {
+        todo.id = uuid();
+        todo.done = 'false';
+        todo.description = todo.description.trim();
+        readTodos()
+          .then(todos => {
```
Comment (gajduk) [inline week3/homework/src/actions/createTodo.js]: Good use of promises 👍

File: week3/homework/src/actions/deleteTodo.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const { readTodos, saveTodos } = './index';
+
+function deleteTodo(req, res) {
+  return new Promise((resolve, reject) => {
+    readTodos()
+      .then(todos => {
+        const newTodos = todos.filter(todo => todo.id !== req.params.id);
```
Comment (gajduk) [inline week3/homework/src/actions/deleteTodo.js]: nice 👍

File: week3/homework/src/actions/readTodo.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+function readTodo(req, res) {
+  return new Promise((resolve, reject) => {
+    fs.readFile(path, 'utf8', (error, todos) => {
+      if (error) {
+        reject(error);
+      }
+      else {
+        const parsedTodos = JSON.parse(todos);
+        const todo = parsedTodos.filter(todo => todo.id === req.params.id);
```
Comment (gajduk) [inline week3/homework/src/actions/readTodo.js]: I am curious. Why do you sometimes use `filter `and other times `find `?

File: week3/homework/src/actions/updateTodo.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+const { readTodos, saveTodos } = './index';
+
+function updateTodo(req, res) {
+  return new Promise((resolve, reject) => {
+    readTodos()
+      .then(todos => {
+        const todoItem = todos.find(todo => todo.id === req.params.id);
```
Comment (gajduk) [inline week3/homework/src/actions/updateTodo.js]: Deja vu? Remember to keep your code DRY. When some code is repeating over and over again it is a signal that it wants to be in a separate function.

File: week3/homework/src/index.js [hunk 7]
```diff
@@ -1,3 +1,50 @@
...
+  createTodo(req, res);
+});
+
+app.get('/todos', (req, res) => {
+  readTodos(req, res)
+    .then(todos => res.status(200).json(todos))
+    .catch(error => res.status(500).send("can't read file!", error));
+});
+
+app.put('/todos/:id', (req, res) => {
```
Comment (gajduk) [inline week3/homework/src/index.js]: very pleasant to read

File: week3/homework/src/actions/clearTodos.js (hunk 1 above)
Comment (Nadoshka) [reply week3/homework/src/actions/clearTodos.js]: Thanks! u are right! It's done!

File: week3/homework/src/index.js (hunk 7 above)
Comment (Nadoshka) [reply week3/homework/src/index.js]: Thanks a lot for your feedback! It's very appreciated! :)

File: week3/homework/src/actions/readTodo.js (hunk 5 above)
Comment (Nadoshka) [reply week3/homework/src/actions/readTodo.js]: No particular reason! Do u think is it best practice to use one than the other? 
Thanks in advance for your feedback!

File: week3/homework/src/actions/updateTodo.js (hunk 6 above)
Comment (Nadoshka) [reply week3/homework/src/actions/updateTodo.js]: Thanks for your feedback! I will keep it in mind!

File: week3/homework/src/actions/deleteTodo.js (hunk 4 above)
Comment (Nadoshka) [reply week3/homework/src/actions/deleteTodo.js]: Thanks a lot! :)

File: week3/homework/src/actions/createTodo.js (hunk 3 above)
Comment (Nadoshka) [reply week3/homework/src/actions/createTodo.js]: Thanks!

File: week3/homework/src/actions/readTodo.js (hunk 5 above)
Comment (gajduk) [reply week3/homework/src/actions/readTodo.js]: Not so much best practice, but being consistent is always preferred. If you are looking for exactly one element I think find is more clear than filter.

File: week3/homework/src/actions/updateTodo.js (hunk 6 above)
Comment (gajduk) [reply week3/homework/src/actions/updateTodo.js]: function findTodoItemById(todos) {
    const todoItem = todos.find(todo => todo.id === req.params.id);
    if (!todoItem) throw new Error('item not found!'));
    else return todoItem;
}

File: week3/homework/src/index.js [hunk 8]
```diff
@@ -13,38 +13,63 @@ const {
...
-  updateTodo();
+  updateTodo(req, res)
+    .then(todos => res.status(200).json(todos))
+    .catch(error => res.status(500).send({ message: error.message }));
 });
 
 app.delete('/todos/:id', (req, res) => {
-  deleteTodo();
+  deleteTodo(req, res)
+    .then(todo => res.status(200).json(todo))
```
Comment (gajduk) [inline week3/homework/src/index.js]: Even though there is more code now, I think it is a huge improvement with good error handling and nice error messages. Good job 👍

**Pull request closed without merging**